### PR TITLE
dashboard: detect nonexistent profile instead of causing a panic

### PR DESCRIPTION
**Pull request by fenglixa**
Fix for #4359

My UT result could see the picture:

image

### CODE REVIEW

File: cmd/minikube/cmd/dashboard.go [hunk 1]
```diff
@@ -79,6 +71,28 @@ var dashboardCmd = &cobra.Command{
 		if err != nil {
 			exit.WithError("Error getting client", err)
 		}
+
+		_, err = api.Load(pkg_config.GetMachineName())
+		if err != nil {
+			switch err := errors.Cause(err).(type) {
+			case mcnerror.ErrHostDoesNotExist:
+				console.OutStyle(console.Meh, "%q cluster does not exist", pkg_config.GetMachineName())
+				os.Exit(0)
```
Comment (tstromberg) [inline cmd/minikube/cmd/dashboard.go]: I think this should still be a non-zero exit. Perhaps exit.Unavailable?

File: cmd/minikube/cmd/dashboard.go [hunk 2]
```diff
@@ -79,6 +71,28 @@ var dashboardCmd = &cobra.Command{
 		if err != nil {
 			exit.WithError("Error getting client", err)
 		}
+
+		_, err = api.Load(pkg_config.GetMachineName())
+		if err != nil {
```
Comment (tstromberg) [inline cmd/minikube/cmd/dashboard.go]: Join these two expressions:

`if _, err = api.Load(pkg_config.GetMachineName()); err != nil {`

**Comment by tstromberg** [comment]
Thank you for doing this! This is a much better user experience.

File: cmd/minikube/cmd/dashboard.go (hunk 1 above)
Comment (fenglixa) [reply cmd/minikube/cmd/dashboard.go]: Thanks @tstromberg, done.

File: cmd/minikube/cmd/dashboard.go (hunk 2 above)
Comment (fenglixa) [reply cmd/minikube/cmd/dashboard.go]: Thanks @tstromberg, done. New UT result:
image

**Comment by medyagh** [comment]
@minikube-bot OK to test

**Comment by sharifelgamal** [comment]
@minikube-bot OK to test

**Pull request merged**